3696/1/top_gems.rb

```diff
@@ -0,0 +1,66 @@
+require 'yaml'
+require 'open-uri'
+require 'optparse'
+require './table_drawer'
+require './scraper'
+require './name_to_link_transformer.rb'
+
+class TopGems
+  def initialize
+    @options = parse
+  end
+
+  def run
+    TableDrawer.new(ranking(gem_names, name_pattern), top).draw
+  end
+
+  private
+
+  attr_reader :options
+
+  def ranking(file, name)
+    file.each_with_object({}) do |str, hsh|
+      next unless str.start_with? name
+
+      transformer_link = NameToLinkTransformer.new(str).link
+      hsh[str] = scrape_value(transformer_link) unless transformer_link == 'no-link'
+    end
+  end
+
+  def gem_names
+    @gem_names ||= begin
+      path = options[:file] || 'example.yaml'
+      YAML.load_file(path)['gems']
+    end
+  end
+
+  def name_pattern
+    @name_pattern ||= @options[:name] || ''
+  end
+
+  def top
+    @top ||= @options[:top].to_i
+  end
+
+  def scrape_value(link)
+    scraper = Scraper.new(link)
+    info = scraper.scrape
+    scraper.close
```

Comment:
look at how writing into file looks like in ruby. It provides block and closes the file automatically in the end. Actually I won't mind if you leave it as is or move closing into the `scrape` method